### PR TITLE
Run test_mps_allocator_module serially

### PR DESCRIPTION
**Pull request by huydhn**
Not sure why this test starts to fail (maybe runner update) https://hud.pytorch.org/hud/pytorch/pytorch/8a2fed7e6ab4cb6a97d92f6ca14f257370ec3a92/1?per_page=50&name_filter=mps or why it was XFAIL in this old PR https://github.com/pytorch/pytorch/pull/97151, but the test is passing locally for me now

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

**Comment by huydhn** [comment]
@pytorchbot revert -m 'The test is now failing again in trunk after a day or so of staying green, we need to continue the investigation' -c weird

An example failure https://hud.pytorch.org/pytorch/pytorch/commit/dd00f5e78d44f55595728ef03018c01296a31ec9

**Comment by Flamefire** [comment]
> @pytorchbot revert -m 'The test is now failing again in trunk after a day or so of staying green, we need to continue the investigation' -c weird

Yes especially the use of `assertTrue` instead of more specific tests make it very hard to investigate the failures.

The test looks very strange though:
```
self.assertTrue(current_alloc_before == 0)
...
self.assertTrue(current_alloc_after > current_alloc_before)
```

If the "before" must be zero, why the use in the 2nd assert? I'd say the first assert is the issue: Whether or not `gc.collect()` actually collects everything is up to the runtime (and moon phase ;-) )

**Comment by malfet** [comment]
> If the "before" must be zero, why the use in the 2nd assert?

Because `current_alloc_after` could return zero (and I guess that what this test was trying to do)

**Comment by malfet** [comment]
> @pytorchbot revert -m 'The test is now failing again in trunk after a day or so of staying green, we need to continue the investigation' -c weird
> 
> An example failure https://hud.pytorch.org/pytorch/pytorch/commit/dd00f5e78d44f55595728ef03018c01296a31ec9

It's weird, but it is not, this test is just incompatible with parallel run mode, as it assumes now MPS allocation happens between those lines:
https://github.com/pytorch/pytorch/blob/92be3403ea28c1c5a71593161fd9704c0fd3b451/test/test_mps.py#L7835-L7838

But if another test runs in parallel, we can not ensure that memory allocation would be zero at that point. Let's just add `unittest.skip` for this one, and create a follow up issue on who to run any memory measurement tests (I think for CUDA we run those in serialized mode)

**Comment by Flamefire** [comment]
> Because `current_alloc_after` could return zero (and I guess that what this test was trying to do)

No I meant if you assert `current_alloc_before=0` then why not just `self.assertTrue(current_alloc_after > 0)` after?

File: test/test_mps.py [hunk 1]
```diff
@@ -7829,7 +7829,7 @@ def test_device_synchronize(self):
         x.backward(torch.randn_like(x))
         torch.mps.synchronize()
 
-    @unittest.expectedFailure
+    @serialTest
```
Comment (clee2000) [inline test/test_mps.py:7832]:
```suggestion
    @serialTest()
```

Comment (huydhn) [reply test/test_mps.py:7832]: Ouch, this makes me think I need to to check if the test is really running

**Review by malfet — APPROVED** [review]
LGTM, but please undo kineto submodule update
